### DISCUSSION TITLE
Connection refused

### DISCUSSION THREAD

**Discussion by coxmaul** [marcone/teslausb#900] [Q&A]
I successfully connected raspberry pi zero w  to wifi after editing "teslausb_setup_variables.conf.sample"
 Having a problem with ssh to pi@'myComputerIP'
Used the password 'raspberry'.
And received 'connection refused'

<img width="469" alt="Screenshot 2024-11-22 at 9 27 23 PM" src="https://github.com/user-attachments/assets/ea824a12-0525-4da6-a9af-dc7dd902b86f">

What can i do to fix this?

**Comment by jjermany** [comment]
ssh pi@teslausb.local instead

**Reply by coxmaul** [reply]
That was my first attempt. I wasn’t getting any response from that.

**Reply by jjermany** [reply]
It sounds like it's not connecting to your home network. I would recheck the ssid and password settings and give it a fresh installation. Also note that the raspberry pi w only supports 2.4ghz WiFi.

**Reply by coxmaul** [reply]
Raspberry pi shows up on my router network. It should be connected i think and I did use 2.4.

**Comment by coxmaul** [comment]
I created a ssh text file in the boot folder. Still no luck

**Comment by marcone** [comment]
> I successfully connected raspberry pi zero w to wifi after editing "teslausb_setup_variables.conf.sample" Having a problem with ssh to pi@'myComputerIP' Used the password 'raspberry'. And received 'connection refused'

How do you know that the Pi successfully connected to wifi, and what are you specifying as "myComputerIP"? You need to specify the IP address of the Pi, not the IP address of the computer you're using to ssh from.
If you put the SD card back in the computer you can check the setup log to see if setup completed or something went wrong.

**Reply by coxmaul** [reply]
Cause pi shows up in my router network. 
The video instruction was showing that My main computer ip needs to be specified. I thought it was because the videos from Tesla are exported to my main computer. Checking the set up log from teslausb_setup_variables.conf?

**Reply by marcone** [reply]
> The video instruction was showing that My main computer ip needs to be specified

I don't think so. The IP or name of your archive server goes in the config file in the `ARCHIVE_SERVER` variable, but to ssh into the Pi you need to use `ssh pi@<IP address of the Pi>`. If your router shows the Pi is on the network, then it will probably also show the IP address of the Pi. Use that IP in the `ssh` command.

> Checking the set up log from teslausb_setup_variables.conf?

That does not make any sense. I meant: take the sd card out of the Pi, and then connect that sd card to your computer/laptop like you did when you flashed it. A drive named `boot` should appear. The TeslaUSB setup log will be in that "boot" drive.

**Reply by coxmaul** [reply]
i tried ssh pi@<IP address of the Pi>. now im getting Permission Denied.
Also tried sudo ssh pi@<IP address of the Pi>. This comes out Sorry,please try again.
I wonder what's causing the permission issue.

**Reply by coxmaul** [reply]
I reinstalled pi os and tried again.

I got.

IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ED25519 key sent by the remote host is
SHA25sldfjeliajsfeliajfsdkjsadfsadfasfL4sLWk.
Please contact your system administrator.
Add correct host key in /var/root/.ssh/known_hosts to get rid of this message.
Offending ED25519 key in /var/root/.ssh/known_hosts:1
Host key for 192.168.1.27 has changed and you have requested strict checking.
Host key verification failed.

Then I tried sudo ssh pi@<IP address of the Pi>. It asked for a password. 
Entered raspberry
I got, Sorry try again.

**Reply by marcone** [reply]
That scary looking warning is a good sign at least. It means you're actually connecting to the Pi.
Easiest way to solve the new "problem" is to delete the file "/var/root/.ssh/known_hosts" before running the ssh command.

> Then I tried sudo ssh pi@. It asked for a password.
> Entered raspberry
> I got, Sorry try again.

That's because it was asking for your Macbook's sudo password, not the Pi's login password.

**Reply by coxmaul** [reply]
this guy mentioned about deleting that file. https://www.youtube.com/watch?v=ETs6r1vKTO8 @22.38sec
I ran his command and your command. known_hosts or notepad known_hosts
I always get zsh: command not found: known_hosts
OR
command not found: notepad

**Reply by marcone** [reply]
"known_hosts" is not a command, it's the name of the file that you need to delete. "notepad" is the name of a Windows utility. Based on your earlier screenshot, you're using a Macbook, so not running Windows.
Try `rm /var/root/.ssh/known_hosts`, then `ssh pi@192.168.1.27`

**Reply by coxmaul** [reply]
The folder name is bootfs on my end, not boot. i don't know if that matters.
'rm /var/root/.ssh/known_hosts' gets me 'permission denied'

**Reply by marcone** [reply]
> The folder name is bootfs on my end, not boot. i don't know if that matters.

That shouldn't matter. The TeslaUSB setup log should be in that bootfs folder. Did you check whether it's there and what it contains?

> 'rm /var/root/.ssh/known_hosts' gets me 'permission denied'

Then you need to run `sudo rm /var/root/.ssh/known_hosts` (and then enter your Macbook account login password, not "raspberry").

**Reply by coxmaul** [reply]
<img width="555" alt="Screenshot 2024-11-23 at 6 54 37 PM" src="https://github.com/user-attachments/assets/e90fc2c4-3baa-4cc1-b879-3abc30808e6b">
I think i got one step further. Entered 'pi@192.168.1.27' , it asked for password. 
I tried both mac password and raspberry password and it's saying permission denied.
I did this step after sudo rm /var/root/.ssh/known_hosts
Feels like almost there :(

**Reply by marcone** [reply]
And you're sure that 192.168.1.27 is still the IP address of the Pi? (it can change across reboots)
What image did you flash onto the Pi?

**Reply by coxmaul** [reply]
Yes thats the ip address. 
i used Pi zero os with Bullseye

**Reply by marcone** [reply]
> i used Pi zero os with Bullseye

Do you mean the prebuilt TeslaUSB image at https://github.com/marcone/teslausb/releases, or did you use the Raspberry Pi Imager to select an official Raspberry Pi OS image?

**Reply by coxmaul** [reply]
i didn't know there is a prebuilt teslamate image. 
So this time i used raspberry pi imager to select Pi zero OS other, lite.
And then ran Raspberry pi imager again with Teslausb.img Under Custom for Operating system.
(In the past, i was using Raspberry pi os bullseye instead of pi os(other) lite)

For some reason,  Teslausb.img wasn't replacing the previous Pi OS image rather it added a config sample file. 

Ran ssh pi@192.168.1.27
i got connection refused

**Reply by marcone** [reply]
I'm really confused as to what you're actually doing. If previously you flashed an official Raspberry Pi OS image, then you would have had to set up networking and users manually (you can do this from the Raspberry Pi Imager). It sounds like maybe you did that (since the Pi was on the network), but then you either didn't specify a user, or you set up a user that wasn't named "pi", or you specified a password that was not "raspberry". That would explain why logging in with the pi/raspberry login didn't work.

**Reply by coxmaul** [reply]
im confused too :(
On pi imager , OS customization, under "set username and password", i put my username and password. SSID info is entered.

In Teslausb_setup_variables. conf, i also put my SSID info.

I just found this in the log.
Thu 12 Oct 15:21:29 EDT 2023 : Detecting whether to update wpa_supplicant.conf
Thu 12 Oct 15:21:29 EDT 2023 : Wifi variables specified, and no /boot/WIFI_ENABLED. Building wpa_supplicant.conf.

I'd love to chat 1v1 instead of posting, if you have time.

**Reply by marcone** [reply]
> On pi imager , OS customization, under "set username and password", i put my username and password. SSID info is entered

Then that is the username and password you should use when ssh'ing into the Pi. For example if you specified username "nomnom" in the OS customization screen, you would use `ssh nomnom@192.168.1.27` and enter the password you set in the OS customization screen.

**Reply by coxmaul** [reply]
I enter ssh 'user'@192.168.1.27
I get, ssh: connect to host 192.168.1.27 port 22: Connection refused

Strangely, i enter,  sudo ssh 'user'@192.168.1.27
Asks for password. I put the password that i saved on Raspberry Pi imager and it says Sorry, Try again.
I put my macbook password and it gives, ssh: connect to host 192.168.1.27 port 22: Connection refused

Do we have anything to take away from this log? 
Thu 12 Oct 15:21:29 EDT 2023 : Detecting whether to update wpa_supplicant.conf
Thu 12 Oct 15:21:29 EDT 2023 : Wifi variables specified, and no /boot/WIFI_ENABLED. Building wpa_supplicant.conf.

**Reply by coxmaul** [reply]
i am able to ping 192.168.1.27

**Reply by marcone** [reply]
> I enter ssh 'user'@192.168.1.27 I get, ssh: connect to host 192.168.1.27 port 22: Connection refused
> 
> Strangely, i enter, sudo ssh 'user'@192.168.1.27 Asks for password.

This is likely because you just (re)started the Pi and it wasn't yet connected to the wifi when you first typed the ssh command.
Using `sudo ssh user@...` instead of just `ssh user@...` doesn't really do anything except run the ssh command as root on your Macbook. It doesn't change anything as far as the actual ssh login into the Pi goes. So I would suggest not using `sudo` unless it's actually necessary, because you're just unnecessarily complicating the issue.

At this point it's probably easier to start over from scratch, and you need to start by picking one of two installation methods. Either flash the [prebuilt TeslaUSB image](https://github.com/marcone/teslausb/releases/tag/v3.2) and follow the "[one step setup](https://github.com/marcone/teslausb/blob/main-dev/doc/OneStepSetup.md)", _**OR**_ flash one of the official Raspberry Pi OS images and use the [manual setup](https://github.com/marcone/teslausb/wiki/Installation).

Don't mix and match. Follow one process or the other, and follow the steps for that process precisely.

**Reply by coxmaul** [reply]
man this is crazy. i tried from the beginning step by step. 
Im keep getting connection refused after entering the password.-_-

**Reply by marcone** [reply]
> Im keep getting connection refused after entering the password.-_-

Getting a "connection refused" message _**after**_ entering the password makes no sense at all. Unless you're still using `sudo ssh`, which you shouldn't.

**Reply by coxmaul** [reply]
I’m not using sudo ssh. 
i guess I’ll have to buy adapters to connect it to a monitor..

**Reply by marcone** [reply]
Also:

> i didn't know there is a prebuilt teslamate image.

"Teslamate" is an entirely different project, and if that's what you're trying to set up then you're in the wrong discussion forum here.

**Reply by coxmaul** [reply]
Oh i meant teslausb. I just did teslamate before this. I had a problem with SSH with this project too, so I had to buy an adapter for the monitor. I’m using a different raspberry pi so I need different adapters again

**Reply by marcone** [reply]
And also, you said you're using a Raspberry Pi Zero W, which is a bit slow to boot, so you may simply need to wait a little longer after powering it on before it'll actually accept ssh connections.

**Reply by coxmaul** [reply]
i connected it to a monitor. It automatically ran all the commands and took me to log in and password command 
Entered the info and everything worked right away lol.
i rand the upgrade and update commands.
How can i check It contains everything it needs to contain and ready to be connected to the car?
In a video i watched, he was installing Grafana on Pi. Is this required?

*i have a macbook so i don't have the shared folders. created TeslaCam and TeslaMusic folders on desktop

**Reply by marcone** [reply]
I don't know what you mean by "It automatically ran all the commands". Which commands? What did you actually do?

If you installed TeslaUSB, there should be a setup log in /boot which ends with "All done" if it completed successfully, the green LED on the Pi should be doing a "double blink" every second, and if you connect the Pi to your macbook or another desktop computer, it should be showing a drive labeled "CAM" once the Pi finishes booting up.
If that's not the case, then you either didn't set up TeslaUSB, or setup failed for some reason. In that case I recommend you ignore the video you watched and instead follow the steps described in the TeslaUSB documentation precisely.

TeslaUSB does not require Grafana.

**Reply by coxmaul** [reply]
<img width="659" alt="Screenshot 2024-11-24 at 8 23 42 PM" src="https://github.com/user-attachments/assets/ef587c46-e6b5-4e65-970e-fb50d3a9aa34">
<img width="553" alt="Screenshot 2024-11-24 at 8 23 05 PM" src="https://github.com/user-attachments/assets/d3f3aeeb-b278-4c55-9abd-f0193926c2dd">

![IMG_4365](https://github.com/user-attachments/assets/b7042013-3710-46a5-b09b-249c27cc3a74)
![IMG_4363](https://github.com/user-attachments/assets/9daea1b9-e853-4754-b813-ec3c2ff6eff2)
This is the script showed up automatically when i plugged in the SD card to pi
I'm currently stuck after this command for 20 mins. tail -f /boot/teslausb-headless-setup.log
I put in ssid and wifi pass in setup config file and the pi imager but not seeing Teslausb_setup_started 
Wifi-enabled commands files.
Any idea why?

**Reply by marcone** [reply]
Which image did you flash this time?
Your pictures seem to be showing that setup created wpa_supplicant.conf, but then didn't reboot afterwards like it's supposed to (or perhaps it did and you just didn't capture that in a picture?)
In any case, since you have a login prompt, log in to the Pi and run `ifconfig`. That should show whether you have a wifi connection.
Also run `systemctl status ssh.service` like it shows in that last picture to see why the ssh server isn't starting.

**Reply by coxmaul** [reply]
I ran Teslausb prebuilt image. 
it didnt reboot. It got stuck there for 30 min. 
![IMG_4366](https://github.com/user-attachments/assets/55f38036-ade5-462b-9adc-6183af4da094)
I dont think im supposed to see failed to start /etc/rc.local

**Reply by marcone** [reply]
OK, so it has a wifi connection at least. What does `tail /boot/teslausb-headless-setup.log` say at this point?
And what about `journalctl -xu ssh.service` ?

**Reply by coxmaul** [reply]
![IMG_4368](https://github.com/user-attachments/assets/14ba82af-8b6c-4fb0-8c2a-6e73a07aff1f)
![IMG_4367](https://github.com/user-attachments/assets/4e051316-bf4d-4dd4-b087-0fc7d37a7db1)

**Reply by marcone** [reply]
None of this makes any sense at all, but in the interest of getting you unstuck, you could try `touch /boot/WIFI_ENABLED` and then run `/etc/rc.local`

**Reply by marcone** [reply]
oh, I just noticed you're running the "pre-release" prebuilt image. You might want to try the release image instead: https://github.com/marcone/teslausb/releases/tag/v3.2

**Reply by coxmaul** [reply]
seems like the wifi issue has been worked out with this image ! https://github.com/marcone/teslausb/releases/tag/v3.2 
Not sure why the latest date is the pre-release. I was thinking the latest, the better.
*pi is blinking green about 6 times every sec. blinks fast so i can't count

Now i'm at this point. 
![IMG_4371](https://github.com/user-attachments/assets/391a0850-bb99-461d-949b-4670b78cd6ee)
![IMG_4369](https://github.com/user-attachments/assets/8472b72e-477a-4b87-a498-7ff23c24c4ed)


*side question, what's the difference between 
tail /boot/teslausb-headless-setup.log
what does tail mean?

**Reply by marcone** [reply]
> *side question, what's the difference between
> tail /boot/teslausb-headless-setup.log
> what does tail mean?

The difference between "tail /boot/teslausb-headless-setup.log" and what?
`tail` is a command that prints the last lines of the given text file. Use `man tail` to see its documentation.

**Comment by marcone** [comment]
It looks like maybe you didn't specify the correct user name or password for the file share you created on the archive server. 
Also note that if your archive server is an Apple desktop or laptop, you have to specifically set the file share to use Windows-style sharing, not Apple file sharing. I'm not sure what the setting looks like in current versions of MacOS, but in an older version it was an option called "Share files and folders using SMB".

**Reply by coxmaul** [reply]
changed the name to my macbook here. export SHARE_USER=nomnom
Ran journalctl -xu ssh.service
![IMG_4373](https://github.com/user-attachments/assets/bdc5307e-84cc-4c26-82e8-ec8075cefbb2)

"Share files and folders using SMB" this is right.
![IMG_4375](https://github.com/user-attachments/assets/46c0d8e0-4220-4e7a-8283-ec2df00d4d91)
![IMG_4374](https://github.com/user-attachments/assets/f16d5689-2759-4636-a782-5febfbd3dacb)

Did i make any progress?

**Reply by marcone** [reply]
> Did i make any progress?

Well, the settings on the MacBook seem OK, you just need to reboot the Pi so setup can continue there.
It also looks like ssh should be working now, so try ssh'ing into the Pi.

**Reply by coxmaul** [reply]
After I took the screenshots, i rebooted and got this.

![IMG_4377](https://github.com/user-attachments/assets/efe9be87-cf90-4c28-93a2-72bccf758ba4)

Im using 64gb SD card so instead of putting a specific GB, ill put %. 
Like CAM_size 80% and  MUSIC_size 20%
———

i tried removing and adding # in front of export MUSIC_SIZE=4G. Still music folder/file is not created. 
<img width="581" alt="Screenshot 2024-11-25 at 6 16 58 AM" src="https://github.com/user-attachments/assets/a3052307-7881-468c-93a6-873e9ebc9549">

Also, i think i'm supposed to see
/mnt/music and music_disk.bin ?

**Reply by marcone** [reply]
In your first picture you can see it says "Error in /root/teslausb_setup_variables.conf", followed by the error, which is that you accidentally inserted a letter "l" at the start of the first line. You must remove that before continuing. You might want to check if you've accidentally inserted things in other places.

Then you ran the "tail" command in the wrong directory, so it couldn't find the file it was supposed to tail.

You don't have to type `journalctl -xu ssh.service` every time. We know that ssh works now.

TeslaUSB no longer supports percentage-based sizes. You have to specify the exact size. The default 30G is fine, especially since you're using a small SD card (I recommend at least 128 GB).

Did you re-run setup after changing the config file to add the 4GB music file? If you didn't, then nothing will change.

**Reply by coxmaul** [reply]
This time there is no error it seems. But stuck here again.
I deleted the image and redownloaded Teslausb image.
![IMG_4384](https://github.com/user-attachments/assets/186d8762-f459-44fd-95c0-9be7f08a1eb3)

**Reply by marcone** [reply]
It's not able to access the file share for some reason. You'll need to figure out why.

**Reply by coxmaul** [reply]
Im stuck in this step. It's the correct IP address. I think i set up the shared folder on the mac correctly. I tried different ways to set up the shared folder.
![IMG_4541](https://github.com/user-attachments/assets/35ca63a3-6b4f-4bfe-83cf-adbd5f8d7098)
Im out of resources.

**Reply by marcone** [reply]
The screenshots you posted in the Discord chat show a mismatch between the archive server IP address you specified when configuring TeslaUSB (192.168.1.16, as shown in the screenshot above), and the IP address that the archive server (your Macbook, I think?) itself says it has:

![image](https://github.com/user-attachments/assets/c6009127-68e0-439f-9f87-aa6ffadcd251)
